# Bank Loan Prediction

In [1]:
import numpy as ap
import pandas as pd

In [2]:
bank_data = pd.read_csv("DS1_C4_S3_Loan_Data_Challenge.csv")

### 

### TASK - 1 : Load the data into pandas data frame. identify the number of rows and columns and check if there are any null values in the given data.

In [3]:
bank_data

Loan_ID  Gender Married Dependents     Education Self_Employed  \
0    LP001002    Male      No          0      Graduate            No   
1    LP001003    Male     Yes          1      Graduate            No   
2    LP001005    Male     Yes          0      Graduate           Yes   
3    LP001006    Male     Yes          0  Not Graduate            No   
4    LP001008    Male      No          0      Graduate            No   
..        ...     ...     ...        ...           ...           ...   
609  LP002978  Female      No          0      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   
611  LP002983    Male     Yes          1      Graduate            No   
612  LP002984    Male     Yes          2      Graduate            No   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
1               4583             1508.0         128               360   
2               3000                0.0          66               360   
3               2583             2358.0         120               360   
4               6000                0.0         141               360   
..               ...                ...         ...               ...   
609             2900                0.0          71               360   
610             4106                0.0          40               180   
611             8072              240.0         253               360   
612             7583                0.0         187               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
1                 1         Rural           N  
2                 1         Urban           Y  
3                 1         Urban           Y  
4                 1         Urban           Y  
..              ...           ...         ...  
609               1         Rural           Y  
610               1         Rural           Y  
611               1         Urban           Y  
612               1         Urban           Y  
613               0     Semiurban           N  

[614 rows x 13 columns]

In [4]:
bank_data.shape

(614, 13)

In [5]:
bank_data.isnull().sum()

Loan_ID              0
Gender               0
Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

### Interpretation : There is not any null value in given data and there are 614 rows and 13 columns in given bank data.

### 

### TASK - 2 : Display the list of female customers who are self-employed and an income greater than 4000.

In [6]:
female_customers = bank_data[(bank_data["Gender"]=="Female") & (bank_data["Self_Employed"]=="Yes")
                             & (bank_data["ApplicantIncome"]>4000)]
female_customers

Loan_ID  Gender Married Dependents     Education Self_Employed  \
54   LP001186  Female     Yes          1      Graduate           Yes   
113  LP001392  Female      No          1      Graduate           Yes   
353  LP002142  Female     Yes          0      Graduate           Yes   
370  LP002194  Female      No          0      Graduate           Yes   
404  LP002301  Female      No          0      Graduate           Yes   
430  LP002377  Female      No          1      Graduate           Yes   
439  LP002407  Female     Yes          0  Not Graduate           Yes   
463  LP002489  Female      No          1  Not Graduate           Yes   
493  LP002582  Female      No          0  Not Graduate           Yes   
534  LP002731  Female      No          0  Not Graduate           Yes   
561  LP002813  Female     Yes          1      Graduate           Yes   
613  LP002990  Female      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
54             11500                0.0         286               360   
113             7451                0.0         146               360   
353             5500                0.0         105               360   
370            15759                0.0          55               360   
404             7441                0.0         194               360   
430             8624                0.0         150               360   
439             7142                0.0         138               360   
463             5191                0.0         132               360   
493            17263                0.0         225               360   
534            18165                0.0         125               360   
561            19484                0.0         600               360   
613             4583                0.0         133               360   

     Credit_History Property_Area Loan_Status  
54                0         Urban           N  
113               1     Semiurban           Y  
353               0         Rural           N  
370               1     Semiurban           Y  
404               1         Rural           N  
430               1     Semiurban           Y  
439               1         Rural           Y  
463               1     Semiurban           Y  
493               1     Semiurban           Y  
534               1         Urban           Y  
561               1     Semiurban           Y  
613               0     Semiurban           N

## 

### TASK - 3 : Identify those male customers who hold a graduate degree and have the loan status as "Y" (yes).

In [7]:
male_customers = bank_data[(bank_data["Gender"]=="Male") & (bank_data["Education"]=="Graduate") & 
                           (bank_data["Loan_Status"]== "Y")]
male_customers

Loan_ID Gender Married Dependents Education Self_Employed  \
0    LP001002   Male      No          0  Graduate            No   
2    LP001005   Male     Yes          0  Graduate           Yes   
4    LP001008   Male      No          0  Graduate            No   
5    LP001011   Male     Yes          2  Graduate           Yes   
8    LP001018   Male     Yes          2  Graduate            No   
..        ...    ...     ...        ...       ...           ...   
606  LP002961   Male     Yes          1  Graduate            No   
608  LP002974   Male     Yes          0  Graduate            No   
610  LP002979   Male     Yes         3+  Graduate            No   
611  LP002983   Male     Yes          1  Graduate            No   
612  LP002984   Male     Yes          2  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5849                0.0         146               360   
2               3000                0.0          66               360   
4               6000                0.0         141               360   
5               5417             4196.0         267               360   
8               4006             1526.0         168               360   
..               ...                ...         ...               ...   
606             3400             2500.0         173               360   
608             3232             1950.0         108               360   
610             4106                0.0          40               180   
611             8072              240.0         253               360   
612             7583                0.0         187               360   

     Credit_History Property_Area Loan_Status  
0                 1         Urban           Y  
2                 1         Urban           Y  
4                 1         Urban           Y  
5                 1         Urban           Y  
8                 1         Urban           Y  
..              ...           ...         ...  
606               1     Semiurban           Y  
608               1         Rural           Y  
610               1         Rural           Y  
611               1         Urban           Y  
612               1         Urban           Y  

[279 rows x 13 columns]

In [8]:
print("Count of these customers:", male_customers["Loan_ID"].size)

Count of these customers: 279


#### Interpretation : There are 279 male customers who holds graduate degree and have loan status "Y"

### 

### TASK - 4 : Calculate the number of customers who have the loan status as "Y", three or more dependents and a property in rural area. Display all the details of such customers.

In [9]:
bank_data[(bank_data["Loan_Status"]=="Y") & (bank_data["Dependents"] >= "3") & (bank_data["Property_Area"]=="Rural")]

Loan_ID  Gender Married Dependents     Education Self_Employed  \
126  LP001448    Male     Yes         3+      Graduate            No   
213  LP001715    Male     Yes         3+  Not Graduate           Yes   
255  LP001846  Female      No         3+      Graduate            No   
324  LP002065    Male     Yes         3+      Graduate            No   
352  LP002141    Male     Yes         3+      Graduate            No   
376  LP002219    Male     Yes         3+      Graduate            No   
390  LP002255    Male      No         3+      Graduate            No   
391  LP002262    Male     Yes         3+      Graduate            No   
481  LP002536    Male     Yes         3+  Not Graduate            No   
515  LP002659    Male     Yes         3+      Graduate            No   
522  LP002692    Male     Yes         3+      Graduate           Yes   
539  LP002740    Male     Yes         3+      Graduate            No   
610  LP002979    Male     Yes         3+      Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
126            23803                0.0         370               360   
213             5703                0.0         130               360   
255             3083                0.0         255               360   
324            15000                0.0         300               360   
352             2666             2083.0          95               360   
376             8750             4996.0         130               360   
390             9167                0.0         185               360   
391             9504                0.0         275               360   
481             3095                0.0         113               360   
515             3466             3428.0         150               360   
522             5677             1424.0         100               360   
539             6417                0.0         157               180   
610             4106                0.0          40               180   

     Credit_History Property_Area Loan_Status  
126               1         Rural           Y  
213               1         Rural           Y  
255               1         Rural           Y  
324               1         Rural           Y  
352               1         Rural           Y  
376               1         Rural           Y  
390               1         Rural           Y  
391               1         Rural           Y  
481               1         Rural           Y  
515               1         Rural           Y  
522               1         Rural           Y  
539               1         Rural           Y  
610               1         Rural           Y

### 

### TASK - 5 : A new customer, Warner Kim has applied for a loan of 290000. Warner Kim is a graduate and owns a property in an urban area. Check whether Warner Kim would get his loan approved or not.

In [10]:
Warner_kim_apr = bank_data[(bank_data["LoanAmount"]>=290) & (bank_data["Education"]=="Graduate")
                           & (bank_data["Property_Area"]=="Urban")]

In [11]:
Warner_kim_apr["Loan_Status"].value_counts()

Loan_Status
Y    8
N    2
Name: count, dtype: int64

#### Interpretation : Most of the peoples from urban area and having graduate degree who have applied loan amount above 290000, theirs loan has been approved by the bank, so Warner Kim will get his loan approved.

### 

### TASK - 6 : Identify the customers who are not married, are self employed, live in non urban area and have there loan status as "N" . Calculate and display the gender wise count of such customers.

In [12]:
cust = bank_data[(bank_data["Married"]=="No") & (bank_data["Self_Employed"]=="Yes") &
                 (bank_data["Property_Area"]=="Rural") & (bank_data["Loan_Status"]=="N")]

In [13]:
cust["Gender"].value_counts()

Gender
Male      2
Female    1
Name: count, dtype: int64

#### Interpretation : There are 2 Male customers and 1 female customer who are not married live in non urban area and having status as"N"

## 

### TASK - 7 : Calculate and Display average the income of customers based on each area where their property is located(urban, semiurban, rural).

In [14]:
urban_data = bank_data[bank_data["Property_Area"]=="Urban"]
avg_urban_data = urban_data["ApplicantIncome"].mean()
print("Average income of customers from Urban area : ",avg_urban_data)

Average income of customers from Urban area :  5398.247524752475


In [15]:
semiurban_data = bank_data[bank_data["Property_Area"]=="Semiurban"]
avg_semiurban_data = semiurban_data["ApplicantIncome"].mean()
print("Average income of customers from Semiurban area : ",avg_semiurban_data)

Average income of customers from Semiurban area :  5292.261802575107


In [16]:
rural_data = bank_data[bank_data["Property_Area"]=="Rural"]
avg_rural_data = rural_data["ApplicantIncome"].mean()
print("Average income of customers from Rural area : ",avg_rural_data)

Average income of customers from Rural area :  5554.083798882682


### 

### TASK - 8 : List the customers who have a property in a rural or semirban area, applicants income >4000, co-applicants income >1000 and loan status is "N" count the number of such customers.

In [17]:
cust_data = bank_data[(bank_data["Property_Area"]!="Urban") & (bank_data["ApplicantIncome"]>4000) & 
                      (bank_data["CoapplicantIncome"]>1000) & (bank_data["Loan_Status"]=="N")]
(cust_data["Loan_Status"]=="N").value_counts()

Loan_Status
True    23
Name: count, dtype: int64

In [18]:
new_cust_data = cust_data.replace("N","Y")
(new_cust_data["Loan_Status"]=="N").value_counts()
# Number of true values = 0

Loan_Status
False    23
Name: count, dtype: int64

Interpretation : Before loan status N as Y the total count of N is True 23 after replaccing N to Y the count of N is False 23. it means count of N is 0. 

## 

### TASK - 9 : Display the average income and the count of customers who have 3+ dependents, live in an urban area , and have their loan status as "Y".

In [19]:
data = bank_data[(bank_data["Dependents"]=="3+") & (bank_data["Property_Area"]=="Urban") & (bank_data["Loan_Status"]=="Y")]
count = data["Loan_ID"].count()
print("Count of customers : ",count)

Count of customers :  8


In [20]:
avg_income = data["ApplicantIncome"].mean()
print("Average income in Urban area : ",avg_income)

Average income in Urban area :  11045.625


### 

### TASK - 10 : Display the list of customers who live in rural or semiurban areas, with 3+ dependents and have their income greater than the average income of the customers who live in an urban area.

In [28]:
customers = bank_data[(bank_data["Property_Area"]!="Urban") & (bank_data["Dependents"]=="3+") & 
                      (bank_data["ApplicantIncome"]>avg_income)]
customers

Loan_ID Gender Married Dependents Education Self_Employed  \
34   LP001100   Male      No         3+  Graduate            No   
126  LP001448   Male     Yes         3+  Graduate            No   
155  LP001536   Male     Yes         3+  Graduate            No   
324  LP002065   Male     Yes         3+  Graduate            No   
409  LP002317   Male     Yes         3+  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
34             12500             3000.0         320               360   
126            23803                0.0         370               360   
155            39999                0.0         600               180   
324            15000                0.0         300               360   
409            81000                0.0         360               360   

     Credit_History Property_Area Loan_Status  
34                1         Rural           N  
126               1         Rural           Y  
155               0     Semiurban           Y  
324               1         Rural           Y  
409               0         Rural           N

### What do you infer from this task ? can you infer an important parameter that the bank is using to approve loans?

Interpretation : Peoples from Non urban area and who are not self employeed are getting chance to for approved loan.

## 

### Display the average applicant income of those customers who are from non-urban areas and have their loan status as "Y".

In [37]:
avg = customers[(customers["Property_Area"]=="Rural") & (customers["Loan_Status"]=="Y")]

In [38]:
avg_income = avg["ApplicantIncome"].mean()
print("The average applicant income : ",avg_income)

The average applicant income :  19401.5
